examples/gno.land/p/demo/ufmt/ufmt.gno

```diff
@@ -150,10 +150,49 @@ func (p *printer) doPrintf(format string, args []any) {
 			continue
 		}
 
-		verb := sTor[i+1]
+		length := -1
+		precision := -1
+		i++ // skip '%'
+
+		if i < end && sTor[i] >= '0' && sTor[i] <= '9' {
+			start := i
+			for i < end && sTor[i] >= '0' && sTor[i] <= '9' {
+				i++
+			}
+			if i > start {
+				lengthStr := string(sTor[start:i])
```

Comment:
This not memory efficient because a new string needs to be generated each time. 

How about using `strconv.ParseInt` function instead? This could reduce the generating intermediate strings.